devito/operator.py

```diff
@@ -127,11 +127,14 @@ def __init__(self, expressions, **kwargs):
     @cached_property
     def _argument_defaults(self):
         """
-        Derive all default values from parameters and ensure uniqueness..
+        Derive all default values from parameters and ensure uniqueness.
         """
         default_args = ArgumentMap()
         for p in self.input:
             default_args.update(p.argument_defaults())
+        for p in self.dimensions:
+            if p.is_Sub:
+                default_args.update(p.argument_defaults(default_args))
```

Comment:
Was this not covered in the arguments PR?

Reply:
no because ``SubDimension`` is a new type introduced in this PR. I think I have a way of improving this bit of code, but I'll need another PR for that